worlds/stardew_valley/__init__.py

```diff
@@ -121,18 +120,32 @@ def force_change_options_if_incompatible(self):
         goal_is_perfection = self.options.goal == Goal.option_perfection
         goal_is_island_related = goal_is_walnut_hunter or goal_is_perfection
         exclude_ginger_island = self.options.exclude_ginger_island == ExcludeGingerIsland.option_true
+
         if goal_is_island_related and exclude_ginger_island:
             self.options.exclude_ginger_island.value = ExcludeGingerIsland.option_false
             goal_name = self.options.goal.current_key
             player_name = self.multiworld.player_name[self.player]
             logger.warning(
                 f"Goal '{goal_name}' requires Ginger Island. Exclude Ginger Island setting forced to 'False' for player {self.player} ({player_name})")
+
         if exclude_ginger_island and self.options.walnutsanity != Walnutsanity.preset_none:
             self.options.walnutsanity.value = Walnutsanity.preset_none
             player_name = self.multiworld.player_name[self.player]
             logger.warning(
                 f"Walnutsanity requires Ginger Island. Ginger Island was excluded from {self.player} ({player_name})'s world, so walnutsanity was force disabled")
 
+        if goal_is_perfection and self.options.accessibility != Accessibility.option_full:
+            self.options.accessibility.value = Accessibility.option_full
+            player_name = self.multiworld.player_name[self.player]
```

Comment:
You can use this now
```suggestion
            player_name = self.player_name
```